src/bacnet/basic/object/ai.c

```diff
@@ -120,16 +125,72 @@ void Analog_Input_Init(void)
     }
 }
 
-/* we simply have 0-n object instances.  Yours might be */
-/* more complex, and then you need validate that the */
-/* given instance exists */
+/**
+ * Initialize the analog inputs. Returns false if there are errors.
+ *
+ * @param pInit_data pointer to initialisation values
+ *
+ * @return true/false
+ */
+bool Analog_Input_Set(BACNET_OBJECT_LIST_INIT_T *pInit_data)
+{
+  unsigned i;
+
+  if (!pInit_data) {
+    return false;
+  }
+
+  if ((int) pInit_data->length > MAX_ANALOG_INPUTS) {
+    PRINT("pInit_data->length = %d >= %d", (int) pInit_data->length, MAX_ANALOG_INPUTS);
+    return false;
+  }
+
+  for (i = 0; i < pInit_data->length; i++) {
+    if (pInit_data->Object_Init_Values[i].Object_Instance < BACNET_MAX_INSTANCE) {
+      AI_Descr[i].Instance = pInit_data->Object_Init_Values[i].Object_Instance;
+    } else {
+      PRINT("Object instance %u is too big", pInit_data->Object_Init_Values[i].Object_Instance);
+      return false;
+    }
+
+    if (!characterstring_init_ansi(&AI_Descr[i].Name, pInit_data->Object_Init_Values[i].Object_Name)) {
+      PRINT("Fail to set Object name to \"%s\"", pInit_data->Object_Init_Values[i].Object_Name);
```

Comment:
Consider belt and suspenders in case it is non ASCII garbage it could be a while until you hit \0:
"Fail to set Object name to \"%20s\""
replace 20 with more relevant upper limit of object name

Reply:
tru dat ..... if that happens though we need to correct our DD. Those values do not come from anywhere else. The DEBUG_PRINT will be removed from prod in any way rendering PRINT(...) ; to just ';'